feat: support for session_state

nannany opened the pull request:
refs https://github.com/zitadel/oidc/issues/670

Added support for returning [session_state](https://openid.net/specs/openid-connect-session-1_0.html#:~:text=This%20parameter%20is%3A-,session_state,-Session%20State.%20JSON) in oidc's authentication response.

Attention is paid to the following points: 
- no impact on systems currently using zitadel/oidc
- added session_state when redirected by error [※](https://openid.net/specs/openid-connect-session-1_0.html#:~:text=SHOULD%20also%20return%20the%20Session%20State%20as%20an%20additional%20session_state%20parameter%20in%20the%20Authentication%20Error%20Response)

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

muhlemmer reviewed (commented):
Thanks for the PR. I had a quick look at the standard at https://openid.net/specs/openid-connect-session-1_0.html. If we support `session_state` we are also [required](https://openid.net/specs/openid-connect-session-1_0.html#OPMetadata) to return a `check_session_iframe` in the Discovery response:

> check_session_iframe
>
> REQUIRED. URL of an OP iframe that supports cross-origin communications for session state information with the RP Client, using the HTML5 postMessage API. This URL MUST use the https scheme and MAY contain port, path, and query parameter components. The page is loaded from an invisible iframe embedded in an RP page so that it can run in the OP's security context. It accepts postMessage requests from the relevant RP iframe and uses postMessage to post back the login status of the End-User at the OP.

Can you please add that to the `oidc.DiscoveryConfiguration` struct with an `omitempty` tag. Then implementations can correctly advertise this when doing session management.

https://github.com/zitadel/oidc/blob/eb98343a65d0734071746d8d8ffd48db03207c28/pkg/oidc/discovery.go#L7-L156

nannany commented:
@muhlemmer 
CheckSessionIframe was already implemented in [oidc/pkg/oidc/discovery.go](https://github.com/zitadel/oidc/blob/eb98343a65d0734071746d8d8ffd48db03207c28/pkg/oidc/discovery.go#L32) .

On the other hand, we changed it to return check_session_iframe in the discovery response.
(We have changed the Configuration interface, but we think it is rare for implementors to do this directly.)
https://github.com/zitadel/oidc/pull/712/commits/689d6ed6758d79c5826c33352470db4214d4b0c8
